logical/framework/path.go

```diff
@@ -89,7 +108,82 @@ type Path struct {
 	HelpDescription string
 }
 
-func (p *Path) helpCallback() OperationFunc {
+// OperationHandler defines and describes a specific operation handler.
+type OperationHandler interface {
+	Handler() OperationFunc
+	Properties() OperationProperties
+}
+
+// OperationProperties describes an operation for documentation, help text,
+// and other clients. A Summary should always be provided, whereas other
+// fields can be populated as needed.
+type OperationProperties struct {
+	// Summary is a brief (usually one line) description of the operation.
+	Summary string
+
+	// Description is extended documentation of the operation and may contain
+	// Markdown-formatted text markup.
+	Description string
+
+	// Examples provides samples of the expected request data. The most
+	// relevant example should be first in the list, as it will be shown in
+	// documentation that supports only a single example.
+	Examples []RequestExample
+
+	// Responses provides a list of response description for a given response
+	// code. The most relevant response should be first in the list, as it will
+	// be shown in documentation that only allows a single example.
+	Responses map[string][]Response
```

Comment:
Should this be `map[int][]Response`? `http.Response.StatusCode` in Go is defined as an int.

Reply:
It's a public interface though, so once plugins build to it, if we change it later it will break compilation for all of them. Not a big deal to fix, generally, but not nice. Unless you don't think they'll build to it for some reason?

Reply:
True. Let me revise to: doesn't need to block a beta2 release, but does need to be locked down before an RC. I'm mainly speaking re: the desire to get this merged today, but this mapstructure/json stuff tending to take a little time.

Reply:
I can deal with that 👍